xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,16 +768,15 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
-#end
+#set ($specialClassAttribute = $stringtool.toAlphaNumeric("$!paneldoc.documentReference"))
```

Comment:
I'm wondering if we should strip any trailing `WebHome` or not. In the future page API we won't have `WebHome` in the page reference and in the future the direction is not display `WebHome` so since this is the equivalent of an API maybe we shouldn't have `WebHome` in the class name?

Reply:
@mflorea @evalica @tmortagne WDYT?

Reply:
I agree with stripping the `WebHome`.

Reply:
is there a way to verify if the page is terminal or not? if it's terminal use the old / backward compatible class created with name and if it's not terminal, use the documentReference?

Anyway, in CSS will still be hard to style documentReference since they will contain dots '.' and those will need escaping. Hope the reference will not contain ':'

Reply:
> is there a way to verify if the page is terminal or not? if it's terminal use the old / backward compatible class created with name and if it's not terminal, use the documentReference?

Except that the old name is not good enough since you can have several panel pages with the same name in different spaces.

> Anyway, in CSS will still be hard to style documentReference since they will contain dots '.' and those will need escaping. Hope the reference will not contain ':'

We only keep alpha numeric chars.

Reply:
The `$stringtool.toAlphaNumeric` method removes the dots. To be more precise it will remove accents and keep only letters and digits.